src/main/java/edu/wpi/grip/core/serialization/Project.java

```diff
@@ -0,0 +1,76 @@
+package edu.wpi.grip.core.serialization;
+
+import com.google.common.eventbus.EventBus;
+import com.thoughtworks.xstream.XStream;
+import edu.wpi.grip.core.*;
+import edu.wpi.grip.core.sources.CameraSource;
+import edu.wpi.grip.core.sources.ImageFileSource;
+
+import java.io.*;
+import java.util.Optional;
+
+/**
+ * Helper for saving and loading a processing pipeline to and from a file
+ */
+public class Project {
+
+    private final XStream xstream = new XStream();
+    private final Pipeline pipeline;
+    private Optional<File> file = Optional.empty();
+
+    public Project(EventBus eventBus, Pipeline pipeline, Palette palette) {
+        this.pipeline = pipeline;
+
+        this.xstream.registerConverter(new StepConverter(eventBus, palette));
+        this.xstream.registerConverter(new SourceConverter(eventBus, xstream.getMapper()));
+        this.xstream.registerConverter(new SocketConverter(xstream.getMapper(), pipeline));
+
+        this.xstream.registerConverter(new ConnectionConverter(eventBus));
+        this.xstream.registerConverter(new InjectedObjectConverter<>(eventBus), XStream.PRIORITY_VERY_HIGH);
+        this.xstream.registerConverter(new PythonScriptOperationConverter(), XStream.PRIORITY_VERY_HIGH);
+        this.xstream.registerConverter(new MatConverter());
```

Comment:
Size, point, and scalar are never serialized.

Neither is `Mat`.  That was dead code from the previous serialization implementation, which I've removed now.